typed/remove_test.go

```diff
@@ -0,0 +1,791 @@
+				}
+				exGot := tv.ExtractItems(quadruplet.set)
+				if !value.Equals(exGot.AsValue(), exOut.AsValue()) {
+					t.Errorf("ExtractItems expected\n%v\nbut got\n%v\n",
+						value.ToString(exOut.AsValue()), value.ToString(exGot.AsValue()),
+					)
+				}
+
+			}
+		})
+	}
+}
+
+func TestRemove(t *testing.T) {
+	for _, tt := range removeCases {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			tt.test(t)
+		})
+	}
+}
+
+type reversibleExtractTestCase struct {
+	name         string
+	rootTypeName string
+	schema       typed.YAMLObject
+	pairs        []reversibleExtractPair
+}
+
+type reversibleExtractPair struct {
+	object typed.YAMLObject
+	pso    typed.YAMLObject
+}
+
+var reversibleExtractCases = []reversibleExtractTestCase{{
+	name:         "nested types",
+	rootTypeName: "type",
+	schema:       typed.YAMLObject(nestedTypesSchema),
+	pairs: []reversibleExtractPair{{
+		// add to top level element
+		`{"listOfLists": [{"name": "a", "value": ["b", "c"]}, {"name": "d"}]}`,
+		`{"listOfLists": [{"name": "f", "value": ["j", "k"]},]}`,
+	}, {
+		// add to leaf element
+		`{"listOfLists": [{"name": "a", "value": ["b", "c"]}, {"name": "d"}]}`,
+		`{"listOfLists": [{"name": "a", "value": ["j", "k"]},]}`,
+	}, {
+		// add to top level element
+		`{"listOfMaps": [{"name": "a", "value": {"b":"x", "c":"y"}}, {"name": "d", "value": {"e":"z"}}]}`,
+		`{"listOfMaps": [{"name": "f", "value": {"q":"p"}}]}`,
+	}, {
+		// add to leaf element
+		`{"listOfMaps": [{"name": "a", "value": {"b":"x", "c":"y"}}, {"name": "d", "value": {"e":"z"}}]}`,
+		`{"listOfMaps": [{"name": "a", "value": {"f":"p"}}]}`,
+	}, {
+		// replace leaf element
+		`{"listOfMaps": [{"name": "a", "value": {"b":"x", "c":"y"}}, {"name": "d", "value": {"e":"z"}}]}`,
+		`{"listOfMaps": [{"name": "a", "value": {"b":"p"}}]}`,
+	}, {
+		// add to top level element
+		`{"mapOfLists": {"b":["a","c"], "d":["e", "f"]}}`,
+		`{"mapOfLists": {"x":["y","z"]}}`,
+	}, {
+		// add to leaf element
+		`{"mapOfLists": {"b":["a","c"], "d":["e", "f"]}}`,
+		`{"mapOfLists": {"b":["y","z"]}}`,
+	}, {
+		// add to top level element
+		`{"mapOfMaps": {"b":{"a":"x","c":"z"}, "d":{"e":"y", "f":"w"}}}`,
+		`{"mapOfMaps": {"i":{"j":"k"}}}`,
+	}, {
+		// add to leaf element
+		`{"mapOfMaps": {"b":{"a":"x","c":"z"}, "d":{"e":"y", "f":"w"}}}`,
+		`{"mapOfMaps": {"b":{"j":"k"}}}`,
+	}, {
+		// replace leaf element
+		`{"mapOfMaps": {"b":{"a":"x","c":"z"}, "d":{"e":"y", "f":"w"}}}`,
+		`{"mapOfMaps": {"b":{"a":"k"}}}`,
+	}, {
+		// misc: add another root type
+		`{"listOfMaps": [{"name": "a", "value": {"b":"x", "c":"y"}}, {"name": "d", "value": {"e":"z"}}]}`,
+		`{"mapOfLists": {"b":["y","z"]}}`,
+	}, {
+		// misc: recursive deeply nested leaves
+		`{"mapOfMapsRecursive": {"a":{"b":{"c":null}, "d":{"e":{"f":null}, "g":null}}}}`,
+		`{"mapOfMapsRecursive": {"a":{"d":{"e":{"f":{"q":null}, "p":null}}}}}`,
+	}},
+}}
+
+func filterNonLeaves(set *fieldpath.Set) *fieldpath.Set {
```

Comment:
This sounds like a reasonable implementation to have directly on the `fieldpath.Set` object